### PR TITLE
add CI for --skip_markduplicates

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by maxulysse** [comment]
In the end I changed idea, since the mapped bams have no index, I used the merged bams when skipping markduplicates to restart from the csv file.
I'll be continuing on that theme in this PR.

File: subworkflows/nf-core/mapping.nf [hunk 1]
```diff
@@ -77,14 +79,31 @@ workflow MAPPING {
...
         [meta, bam]
-    }.groupTuple() //groupKey above is somehow makes sure, the workflow doesn't stall until all pieces are mapped, but that the correct group can advance as soon as it is complete
+    }.groupTuple()
     .set{bam_mapped}
 
-    // STEP 1.5: MERGING AND INDEXING BAM FROM MULTIPLE LANES // MarkDuplicates can take care of this
+    // MarkDuplicates can handle multiple BAMS as input, so no merging/indexing at this step
+    // Except if and only if skipping MarkDuplicates
+
+    if (skip_markduplicates) {
```
Comment (FriederikeHanssen) [inline subworkflows/nf-core/mapping.nf:93]: Ah yes, adding it here makes sense. I struggled finding a logical way on how to sort all the different ways how to walk through to base recalibration

File: subworkflows/nf-core/qc_markduplicates.nf [hunk 2]
```diff
@@ -88,12 +79,11 @@ workflow QC_MARKDUPLICATES {
     //if !skip_markduplicates, then QC tools are run on duplicate marked bams
     //After bamqc finishes, convert to cram for further analysis
     qualimap_bamqc = Channel.empty()
-    if(!skip_bamqc){
+    if (!skip_bamqc && !skip_markduplicates) {
```
Comment (FriederikeHanssen) [inline subworkflows/nf-core/qc_markduplicates.nf:82]: should this not be run on the mapped bams if no duplicate marking is done?

**Review by FriederikeHanssen — COMMENTED** [review]
Before approving, just want to briefly discuss when BAMQC should be run. I would have proposed after duplciate marking, if duplicate marking is skipped then after mapping.

File: subworkflows/nf-core/qc_markduplicates.nf (hunk 2 above)
Comment (maxulysse) [reply subworkflows/nf-core/qc_markduplicates.nf:82]: I'll need to do more testing and add them to CI to have all scenario covered.
I figured it was a big enough PR already ;-)

Comment (FriederikeHanssen) [reply subworkflows/nf-core/qc_markduplicates.nf:82]: ok then let's add a TODO here so it won't sneak by unnoticed later

Comment (FriederikeHanssen) [reply subworkflows/nf-core/qc_markduplicates.nf:82]:
```suggestion
    if (!skip_bamqc && !skip_markduplicates) {
    //TODO: after adding CI tests, allow bamqc on mapped bams if no duplicate marking is done
```

**Review by FriederikeHanssen — APPROVED** [review]
